specs/agents/universal-profiling-integration.md

```diff
@@ -173,6 +171,23 @@ All messages have the following layout:
 * *message-type* : An ID uniquely identifying the type (and therefore payload structure) of the message.
 * *minor-version* : The version number for the given *message-type*. This value is incremented when new fields are added to the payload while preserving the *message-type* (non breaking changes). For breaking changes a new *message-type* must be used.
 
+## Profiler Registration Message
+
+Whenever the profiling host agent starts communicating for the first time with a process running an APM Agent, it MUST send this message.
+This message is used to let the APM-agent know that a profiler is actually active on the current host. Note that that an APM-agent may receive this message zero, one or several times: This may happen if no host agent is active, if one is active or if a host agent is restarted during the lifetime of the APM-agent respectively.
+
+The *message-type* is `2` and the current *minor-version* is `1`.
+
+The payload layout is as follows:
+Name                  | Data type 
+--------------------- | -------------
+samples-delay-ms      | uint32
+host-id               | utf8-str
+
+* *samples-delay-ms*: A sane upper bound of the usual time taken in milliseconds by the profiling host agent between the collection of a stacktrace and it being written to the apm-agent via the [messaging socket](#cpu-profiler-trace-correlation-message). The APM-agent will assume that all profiling data related to a span has been written to the socket if a span ended at least the provided duration ago. Note that this value doesn't need to be a hard a guarantee, but it should be the 99% case so that profiling data isn't distorted in the expected case.
+* *host-id*: The [`host.id` resource attribute](https://opentelemetry.io/docs/specs/semconv/attributes-registry/host/) used for the profiling data by this profiling host agent. If an APM-agent is already sending a `host.id` it SHOULD print a warning if the `host.id` is different and otherwise ignore the value received by the host agent. If an agent does not collect the `host.id` by itself, it MUST start sending the `host.id` after receiving it from the profiler host agent to ensure certain correlation features (e.g. cost and CO2 consumption) work correctly.
```

Comment:
> If an APM-agent is already sending a `host.id` it SHOULD print a warning if the `host.id` is different and otherwise ignore the value received by the host agent.

1. This means that dependent calculations such as CO2 and cost calculation will not work / use default values (because of a mismatch in host id). Should we call that out here as a hint?
2. IMHO the APM agent MUST print a warning when this happens. Otherwise this is almost impossible to debug.